### PR TITLE
LowCardinality and Nullable

### PR DESCRIPTION
**Pull request by waralex**
Implementation of `LowCardinality` and `Nullable`. When implementing it, I focused on the Python client. LowCardinality in C++ looks much more complex with a large number of options, but I haven't understood it in much detail yet. 
While debugging `LowCardinality`, I made the table in ClickHouse non-consistent several times and a couple of times ClickHouse fell into the core. It seems that I have fixed everything, but I hope that you will also look at it carefully, since there is a danger of such unpleasant events

### CODE REVIEW

File: src/columns/Nullable.jl [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                                ::Val{:Nullable}, nest::TypeAst) where {T}
+    !can_be_nullable(nest.name) &&
+            error("$(nest.name) cannot be inside Nullable")
+    missing_map = uint8_ismissing.(data)
+    chwrite(sock, missing_map)
+    unmissing = if !any(x -> x > 0, missing_map)
+        convert(CategoricalVector{T}, data)
+    else
+        tmp = deepcopy(data)
+        replace!(tmp.refs, 0=>1)
```
Comment (athre0z) [inline src/columns/Nullable.jl:73]: I think it deserves an explaining comment. I'm guessing 0 is (always?) the missing value of CategoricalArray and in order to remove it, we have to rewrite it so something valid?

Comment (waralex) [reply src/columns/Nullable.jl:73]: Yes, it is

Comment (waralex) [reply src/columns/Nullable.jl:75]: I added comment at bd30b03

File: src/columns/LowCardinality.jl [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    result = is_nullable ?
+            CategoricalVector{Union{T, Missing}}(undef, 0, levels = get.(index))  :
+            CategoricalVector{T}(undef, 0, levels = get.(index))
+    result.refs = keys
+    return result
+end
+
+
+function read_col_data(sock::ClickHouseSock, num_rows::VarUInt,
+                                            ::Val{:LowCardinality}, nest::TypeAst)
```
[the remark this answers is not in the document]
Comment (waralex) [reply src/columns/LowCardinality.jl]: Done at fd6b365

**Review by athre0z — APPROVED** [review]
Thanks, this LGTM now!

**Comment by athre0z** [comment]
Given that you now contributed like half the source code of this library, if this was still my private repo, I feel like it would only be appropriate to ask you whether you'd like to be become a proper maintainer / collaborator of this repo. However, I don't have permissions to do so and also don't really know if this is how things are done in the `JuliaDatabases`  organization. 

@quinnj it would be great if you could help us out here and provide some info on the management of this organization!

@waralex By the way, if you don't want to and rather continue just being an unassociated contributor, that is totally fine as well!

**Comment by waralex** [comment]
@athre0z Of course, I would be happy to become the so-maintainer / collaborator of this repo. However this does not relieve you of the obligation to review my changes :) Just as I am ready to review any changes made by other people

**Comment by athre0z** [comment]
> However this does not relieve you of the obligation to review my changes :) 
Heh, yes, obviously. Any non-trivial changes (trivial = typos, reformatting, ...) would still get reviewed.

**Comment by quinnj** [comment]
@athre0z, I've given you admin privileges (sorry that didn't happen w/ the migration to JuliaDatabases!)
@waralex I've given you maintainer rights.

Love to see all the activity going on here! I opened #12 and assigned myself and I'll try to implement that this week.

**Pull request merged**